rai/node/node.cpp

```diff
@@ -1293,15 +1293,21 @@ bool rai::block_processor::full ()
 	return blocks.size () > 16384;
 }
 
-void rai::block_processor::add (std::shared_ptr<rai::block> block_a, std::chrono::steady_clock::time_point origination)
+void rai::block_processor::add (std::shared_ptr<rai::block> block_a, std::chrono::steady_clock::time_point origination, bool unchecked)
 {
 	if (!rai::work_validate (block_a->root (), block_a->block_work ()))
 	{
 		std::lock_guard<std::mutex> lock (mutex);
 		if (blocks_hashes.find (block_a->hash ()) == blocks_hashes.end ())
 		{
-			blocks.push_back (std::make_pair (block_a, origination));
-			blocks_hashes.insert (block_a->hash ());
+			if (!unchecked && block_a->type () == rai::block_type::state && block_a->link () != node.ledger.epoch_link)
```

Comment:
I believe the `unchecked` parameter is named incorrectly. Shouldn't it be called `checked`? If it's true, it bypasses signature validation.

Reply:
Can we construct a test on that?

Reply:
Well, probably require renaming. Parameter for state blocks from unchecked table. As all state blocks in unchecked already was validated

Reply:
I see. Is the performance improvement worth the risk?

Reply:
I guess yes. Rare risks are signature corruption on disk in unchecked table or bit flips. Same risk for state blocks table & usual signature check.
But this can be moved to separate PR

Reply:
In terms of risk I was thinking we might accidentally put unvalidated blocks in the unchecked table.

Reply:
Though I think I've brought this up before. Either way should be fine.

Reply:
Let’s go with the safer option. Can we make a test that has a block with a bad signature in unchecked and make sure it isn’t processed?

Reply:
Unchecked thing removed. I guess we can return to this after 17.0, now better polish what we have for new release